src/NLog/Common/InternalLoggerMessageEventArgs.cs

```diff
@@ -37,31 +37,48 @@
 namespace NLog.Common
 {
     /// <summary>
-    /// Written to the internal logger
+    /// A message has been written to the internal logger
     /// </summary>
     public sealed class InternalLoggerMessageEventArgs : EventArgs
     {
         /// <summary>
         /// The rendered message
         /// </summary>
         public string Message { get; }
+
         /// <summary>
-        /// The loglevel
+        /// The log level
         /// </summary>
         public LogLevel Level { get; }
 
         /// <summary>
-        /// Optional, the exception
+        /// The exception. Could be null.
         /// </summary>
         [CanBeNull]
         public Exception Exception { get; }
 
+        /// <summary>
+        /// The type that triggered this internal log event, for example the FileTarget. 
+        /// This property is not always populated. 
+        /// </summary>
+        [CanBeNull]
+        public Type ContextType { get; }
```

Comment:
@snakefoot I'm only doubting about these names. (contexttype, contextname)

I see in the original issue https://github.com/NLog/NLog/issues/2710#issuecomment-413304149 it's called SenderType and SenderName. Wouldn't that be better?

Reply:
Excellent names. Please do the honors

Reply:
fixed